File: 프로그래머스/lv2/70129. 이진 변환 반복하기/이진 변환 반복하기.py
def solution(s):
    zero=0
    count=0
    while s!='1':
        zero+=s.count('0')
        s=s.replace('0','')
        s=bin(len(s))[2:]
        count+=1
    answer=[]
    answer.append(count)
    answer.append(zero)
    
    return answer
    
    
        

File: 프로그래머스/2/70129. 이진 변환 반복하기/이진 변환 반복하기.java
class Solution {
    static int count=0;
    public int[] solution(String s) {
        int zero = 0;
        
        while(s.length()>1){
            StringBuilder sb = new StringBuilder();
            char[] arr = s.toCharArray();
            
            for(char ch : arr){
                if(ch=='0'){
                    zero++;
                }else{
                    sb.append(ch);
                }
            }
            String str = sb.toString();            
            s=bin(str.length());
        }
        int[] answer = new int[] {count,zero};
        return answer;
    }
    
    
    public static String bin(int num){
        count++;
        StringBuilder sb = new StringBuilder();
        while (num>0){
            int left=num%2;
            num=num/2;
            sb.append(left);
        }
        return sb.toString();
    }
}